vpr/src/route/check_rr_graph.cpp

```diff
@@ -583,7 +586,9 @@ static bool has_adjacent_channel(const t_rr_node& node, const DeviceGrid& grid)
     /* TODO: this function should be reworked later to adapt RRGraphView interface 
      *       once xlow(), ylow(), side() APIs are implemented
      */
-    VTR_ASSERT(node.type() == IPIN || node.type() == OPIN);
+    auto& device_ctx = g_vpr_ctx.device();
+    const auto& rr_graph = device_ctx.rr_graph;
```

Comment:
We may combine the two lines to one line:
```
const auto& rr_graph = g_vpr_ctx.device().rr_graph;
```

The ``device()`` method will give you a read-only ``DeviceContext`` data structure. So we can use ``const auto&`` rather than a reference like ``auto&``.